app/assets/javascripts/discourse/app/services/composer.js

```diff
@@ -498,6 +506,15 @@ export default class ComposerController extends Controller {
     });
   }
 
+  @action
+  setCategory(categoryId) {
+    this.model.categoryId = categoryId;
+
+    if (this.formTemplateIds?.length) {
+      this.appEvents.trigger("composer:load-templates", this.formTemplateIds);
```

Comment:
`load-templates` feels like a confusing name here. Maybe `selected-form-template` is a little clearer (and more consistent with the `formTemplate` naming in general?

Reply:
Also, related, do you need an app event here? Can we rely on `updateAttrs` maybe (or the Glimmer equivalent)?

Reply:
I've updated the the name to `composer:selected-form-template`.

`updateAttrs` no longer exists in Glimmer, but I couldn't get it to work with `{{did-update}}` or native getters either.

Reply:
what did you try with did-update?

Reply:
I misunderstood, I thought it updates whenever attributes change, so I added it to the `<div>` wrapper and used it to call `_fetchTemplates()` but it appears `{{did-update}}` isn't called when attributes change...

Reply:
Can you just show me the exact syntax you used? {{did-update}} alone is not going to do anything

Reply:
Thanks @pmusaraj @jjaffeux, figured it out:
https://github.com/discourse/discourse/pull/21190/commits/6f9f5d06e87174f91c37308b222f96ebf341876b#diff-3405153053e224658ecfc65f2f4c931b65acf8ea46bbc7b599fad17d463dd1ceR2-R5

The PR no longer uses `appEvents`